proxy/apifilterproxy/manager/filter_manager.go

```diff
@@ -0,0 +1,200 @@
+	Paths          []string `json:"paths"`
+}
+
+//ConfigFileFields stores filter config
+type ConfigFileFields struct {
+	Prefilters   []model.FilterData
+	Destinations []Destination
+}
+
+//SetConfig sets the parameters necessary and initializes API Proxy handler
+func SetConfig(filterConfigFile string, cattleAddr string) {
+	configFile = filterConfigFile
+
+	if configFile == "" {
+		log.Warnf("No path found to the APIfilter config.json file")
+		APIFilterProxyReady = false
+		return
+	}
+
+	CattleURL = "http://" + cattleAddr
+	if len(CattleURL) == 0 {
+		log.Warnf("No CattleAddrset to forward the requests to Cattle")
+		APIFilterProxyReady = false
+		return
+	}
+
+	DefaultDestination = CattleURL
+	refChan := make(chan int, 1)
+	refreshReqChannel = &refChan
+
+	if configFile != "" {
+		ConfigFields = ConfigFileFields{}
+		PathPreFilters = make(map[string][]model.FilterData)
+		PathDestinations = make(map[string]Destination)
+		err := Reload()
+		if err != nil {
+			log.Warnf("Failed to load the API filter proxy config: %v", err)
+			APIFilterProxyReady = false
+			return
+		}
+	}
+
+	APIFilterProxyReady = true
+}
+
+func Reload() error {
+	//put msg on channel, so that any other request can wait
+	select {
+	case *refreshReqChannel <- 1:
+		if configFile != "" {
+			configContent, err := ioutil.ReadFile(configFile)
+			if err != nil {
+				log.Errorf("Error reading config.json file at path %v", configFile)
+				<-*refreshReqChannel
+				return fmt.Errorf("Error reading config.json file at path %v", configFile)
+			}
+			updatedConfigFields := ConfigFileFields{}
+			err = json.Unmarshal(configContent, &updatedConfigFields)
+			if err != nil {
+				log.Errorf("config.json data format invalid, error : %v\n", err)
+				<-*refreshReqChannel
+				return fmt.Errorf("Proxy config.json data format invalid, error : %v", err)
+			}
+
+			updatedPathPreFilters := make(map[string][]model.FilterData)
+			for _, filter := range updatedConfigFields.Prefilters {
+				//build the PathPreFilters map
+				for _, path := range filter.Paths {
+					updatedPathPreFilters[path] = append(updatedPathPreFilters[path], filter)
+				}
+			}
+
+			updatedPathDestinations := make(map[string]Destination)
+			for _, destination := range updatedConfigFields.Destinations {
+				//build the PathDestinations map
+				for _, path := range destination.Paths {
+					updatedPathDestinations[path] = destination
+				}
+			}
+			ConfigFields = updatedConfigFields
+			PathPreFilters = updatedPathPreFilters
+			PathDestinations = updatedPathDestinations
+
+		}
+		<-*refreshReqChannel
+	default:
+		log.Infof("Reload config is already in process, skipping")
+	}
+	return nil
+}
+
+func ProcessPreFilters(path string, api string, body map[string]interface{}, headers map[string][]string) (map[string]interface{}, map[string][]string, string, model.ProxyError) {
+	prefilters := PathPreFilters[path]
+	log.Debugf("START -- Processing pre filters for request path %v", path)
```

Comment:
Unfortunately we don't have a mechanism to change the log level for microservices launched by cattle within the container.